Sources/AblyAssetTrackingInternal/AblyWrapper/DefaultAbly.swift

```diff
@@ -129,38 +132,123 @@ public class DefaultAbly: AblyCommon {
             completion(.success(false))
             return
         }
-        
+
+        do {
+            try performChannelOperationWithRetry(channel: channelToRemove, operation: { channel in
+                guard let presenceData = presenceData else {
+                    return
+                }
+
+                try disconnectChannel(channel: channel, presenceData: presenceData)
+            })
+            
+            channels.removeValue(forKey: trackableId)
+            completion(.success(true))
+        }catch AblyError.connectionError(let errorInfo){
+            completion(.failure(errorInfo.toErrorInformation()))
+        }catch{
+            completion(.failure(ErrorInformation(error: AblyError.connectionError(errorInfo: ARTErrorInfo.create(withCode: 100_000, message: "Unknown error while disconnecting")))))
+        }
+    }
+    
+    private func disconnectChannel(channel:AblySDKRealtimeChannel, presenceData: PresenceData) throws{
+        leavePresence(channel: channel, presenceData: presenceData) { leaveError in
+            if let leaveError = leaveError {
+                throw AblyError.connectionError(errorInfo: leaveError)
+            }
+            
+            channel.unsubscribe()
+            channel.presence.unsubscribe()
+            self.detachFromChannel(channel: channel) { detachError in
+                guard let error = detachError else { return }
+                throw AblyError.connectionError(errorInfo: error)
+            }
+        }
+    }
+    
+    private func leavePresence(channel:AblySDKRealtimeChannel, presenceData: PresenceData?, completion : @escaping (ARTErrorInfo?) throws -> Void){
```

Comment:
I was surprised to see a completion handler with the `throws` modifier. Usually completion handlers are used to inform the caller of a method that the method has completed its work – they're a one-way communication mechanism.

Furthermore, given that the method doesn't ever do anything with any error thrown by `completion`, I don't think that `completion` should have the `throws` modifier.

Reply:
If I'm not mistaken, Swift compiler wouldn't let me pass a throwing closure (function) unless I matched signature of closure using throws. I'll check again and see if I can get rid of this without changing the behaviour